File: Python/underautomation/universal_robots/primary_interface/tool_communication_info_package_event_args.py
import typing
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import ToolCommunicationInfoPackageEventArgs as tool_communication_info_package_event_args

class ToolCommunicationInfoPackageEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = tool_communication_info_package_event_args()
		else:
			self._instance = _internal
	@property
	def tool_communication_is_enabled(self) -> bool:
		return self._instance.ToolCommunicationIsEnabled
	@tool_communication_is_enabled.setter
	def tool_communication_is_enabled(self, value: bool):
		self._instance.ToolCommunicationIsEnabled = value
	@property
	def baud_rate(self) -> int:
		return self._instance.BaudRate
	@baud_rate.setter
	def baud_rate(self, value: int):
		self._instance.BaudRate = value
	@property
	def parity(self) -> int:
		return self._instance.Parity
	@parity.setter
	def parity(self, value: int):
		self._instance.Parity = value
	@property
	def stop_bits(self) -> int:
		return self._instance.StopBits
	@stop_bits.setter
	def stop_bits(self, value: int):
		self._instance.StopBits = value
	@property
	def rx_idle_chars(self) -> float:
		return self._instance.RxIdleChars
	@rx_idle_chars.setter
	def rx_idle_chars(self, value: float):
		self._instance.RxIdleChars = value
	@property
	def tx_idle_chars(self) -> float:
		return self._instance.TxIdleChars
	@tx_idle_chars.setter
	def tx_idle_chars(self, value: float):
		self._instance.TxIdleChars = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/ToolCommunicationInfoPackageEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Tool communication info
	/// </summary>
	public class ToolCommunicationInfoPackageEventArgs : PackageEventArgs {

		/// <summary>
		/// Returns a string representation of the tool communication info.
		/// </summary>
		public override string ToString()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public ToolCommunicationInfoPackageEventArgs()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Is the tool communication interface enabled
		/// </summary>
		[PackageDescription("Is the tool communication interface enabled")]
public bool ToolCommunicationIsEnabled { get; set; }

		/// <summary>
		/// Baud rate for tool serial communication
		/// </summary>
		[PackageDescription("Baud rate")]
public int BaudRate { get; set; }

		/// <summary>
		/// Parity
		/// </summary>
		[PackageDescription("Parity")]
public int Parity { get; set; }

		/// <summary>
		/// Stop bits
		/// </summary>
		[PackageDescription("Stop bits")]
public int StopBits { get; set; }

		/// <summary>
		/// RX Idle Chars
		/// </summary>
		[PackageDescription("RX Idle Chars")]
public float RxIdleChars { get; set; }

		/// <summary>
		/// TX Idle Chars
		/// </summary>
		[PackageDescription("TX Idle Chars")]
public float TxIdleChars { get; set; }
	}
}
